modules/recurring-donations.nix

```diff
@@ -0,0 +1,108 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.recurring-donations;
+  recurring-donations-script = pkgs.writeScript "recurring-donations.sh" ''
+    LNCLI="lightning-cli --lightning-dir=${config.services.clightning.dataDir}"
+    pay_tallycoin() {
+      NAME=$1
+      AMOUNT=$2
+      echo Attempting to pay $AMOUNT sat to $NAME
+      INVOICE=$(torsocks curl -d "satoshi_amount=$AMOUNT&payment_method=ln&id=$NAME&type=profile" -X POST https://api.tallyco.in/v1/payment/request/ | jq -r '.lightning_pay_request') 2> /dev/null
+      if [ -z "$INVOICE" ] || [ "$INVOICE" = "null" ]; then
+        echo "ERROR: did not get invoice from tallycoin"
+        return
+      fi
+      # Decode invoice and compare amount with requested amount
+      DECODED_AMOUNT=$($LNCLI decodepay "$INVOICE" | jq -r '.amount_msat' | head -c 4)
```

Comment:
I'm getting the following error when entering amounts below 1000, for example 500:

```
Apr 19 16:31:02 nix-bitcoin systemd[1]: Starting Run recurring-donations...
Apr 19 16:31:02 nix-bitcoin bash[304]: Attempting to pay 500 sat to djbooth007
Apr 19 16:31:02 nix-bitcoin bash[304]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Apr 19 16:31:02 nix-bitcoin bash[304]:                                  Dload  Upload   Total   Spent    Left  Speed
Apr 19 16:31:17 nix-bitcoin bash[304]: [1.3K blob data]
Apr 19 16:31:17 nix-bitcoin bash[304]: ERROR: requested amount and invoice amount do not match. 500 vs 5000
Apr 19 16:31:17 nix-bitcoin bash[304]: Attempting to pay 500 sat to hillebrandmax
Apr 19 16:31:17 nix-bitcoin bash[304]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Apr 19 16:31:17 nix-bitcoin bash[304]:                                  Dload  Upload   Total   Spent    Left  Speed
Apr 19 16:31:23 nix-bitcoin bash[304]: [553B blob data]
Apr 19 16:31:23 nix-bitcoin bash[304]: ERROR: requested amount and invoice amount do not match. 500 vs 5000
```

Reply:
Yeah, the way the amount is read out from the invoice is totally wrong. Fixing...